Cleanup setup

dbu opened the pull request:
while working on the symfony client, i realized that we are rather implicit in how the symfony cache works.

however, this is a BC break for people that implemented their own HttpCache rather than extending ours. not sure if this can go into 1.x or needs to wait for 2.0 - but even with the trait, we still want this, as there are no typehints on traits.

dbu commented on src/SymfonyCache/CacheEvent.php:
```diff
@@ -39,10 +38,10 @@ class CacheEvent extends Event
+     * @param Request                    $request The request being processed.
      */
-    public function __construct(HttpCache $kernel, Request $request)
+    public function __construct(CacheManipulationInterface $kernel, Request $request)
```
this is the actual BC break. should we just additionally accept a HttpCache without interface until 2.0?

ddeboer replied:
Yes, as it’s a BC break I would prefer to postpone this to 2.0. Should we leave this PR open till then?

dbu replied:
i could remove the typehint and do an instanceof check in the constructor for either HttpCache *or* the interface.

ddeboer replied:
:+1:

And maybe comment it to be something that can be taken out for 2.0?

dbu commented:
ok, adjusted to be BC and also updated the documentation. if you have no better idea for the name of the CacheManipulationInterface, this would be mergeable now.

ddeboer commented on src/SymfonyCache/CacheManipulationInterface.php:
```diff
@@ -0,0 +1,44 @@
+/**
+ * Interface for extra requirements to do refresh operations with the cache.
+ */
+interface CacheManipulationInterface extends HttpKernelInterface
```
What about `CacheInvalidationInterface` or `InvalidationCacheInterface`?

dbu commented:
good idea, renamed the interface to CacheInvalidationInterface.

ddeboer commented:
Cool, thanks!
